x-pack/plugins/security_solution/public/resolver/view/index.tsx

```diff
@@ -3,164 +3,37 @@
-  const hasError = useSelector(selectors.hasError);
-  const relatedEventsStats = useSelector(selectors.relatedEventsStats);
-  const activeDescendantId = useSelector(selectors.uiActiveDescendantId);
-  const terminatedProcesses = useSelector(selectors.terminatedProcesses);
-  const { colorMap } = useResolverTheme();
-
-  useLayoutEffect(() => {
-    dispatch({
-      type: 'userChangedSelectedEvent',
-      payload: { selectedEvent },
-    });
-  }, [dispatch, selectedEvent]);
+  const context = useKibana<StartServices>();
+  const store = useMemo(() => {
+    return storeFactory(context);
+  }, [context]);
 
   return (
-    <StyledResolver
-      data-test-subj="resolverEmbeddable"
-      className={className}
-      backgroundColor={colorMap.resolverBackground}
-    >
-      {isLoading ? (
-        <div className="loading-container">
-          <EuiLoadingSpinner size="xl" />
-        </div>
-      ) : hasError ? (
-        <div className="loading-container">
-          <div>
-            {' '}
-            <FormattedMessage
-              id="xpack.securitySolution.endpoint.resolver.loadingError"
-              defaultMessage="Error loading data."
-            />
-          </div>
-        </div>
-      ) : (
-        <StyledResolverContainer
-          className="resolver-graph kbn-resetFocusState"
-          onMouseDown={onMouseDown}
-          ref={ref}
-          role="tree"
-          tabIndex={0}
-          aria-activedescendant={activeDescendantId || undefined}
-        >
-          {connectingEdgeLineSegments.map(({ points: [startPosition, endPosition], metadata }) => (
-            <EdgeLine
-              edgeLineMetadata={metadata}
-              key={metadata.uniqueId}
-              startPosition={startPosition}
-              endPosition={endPosition}
-              projectionMatrix={projectionMatrix}
-            />
-          ))}
-          {[...processNodePositions].map(([processEvent, position], index) => {
-            const adjacentNodeMap = processToAdjacencyMap.get(processEvent);
-            const processEntityId = entityId(processEvent);
-            if (!adjacentNodeMap) {
-              // This should never happen
-              throw new Error('Issue calculating adjacency node map.');
-            }
-            return (
-              <ProcessEventDot
-                key={index}
-                position={position}
-                projectionMatrix={projectionMatrix}
-                event={processEvent}
-                adjacentNodeMap={adjacentNodeMap}
-                relatedEventsStats={relatedEventsStats.get(entityId(processEvent))}
-                isProcessTerminated={terminatedProcesses.has(processEntityId)}
-                isProcessOrigin={false}
-              />
-            );
-          })}
-        </StyledResolverContainer>
-      )}
-      <StyledPanel />
-      <GraphControls />
-      <SymbolDefinitions />
-    </StyledResolver>
+    <Provider store={store}>
```

Comment:
TODO comment explaining why the `Provider` must be here, a level above resolver.